storage/google/cloud/storage/bucket.py

```diff
@@ -926,6 +927,9 @@ def rename_blob(self, blob, new_name, client=None):
         :rtype: :class:`Blob`
         :returns: The newly-renamed blob.
         """
+        if blob.name == new_name:
+            return blob
+
```

Comment:
@frankyn I wonder if we should still do the copy, and just omit the delete if the names are the same.  I presume that would update the `generation` field, which might or might not be a Good Thing(TM).

@tjcelaya Whichever way it sorts out, we are going to need a new unit test which exercises the new codepath.

Reply:
w.r.t @tseaver comment, the copy operation should still occur though omit the delete request.